### ISSUE TITLE
Is this a normal learning evolution

### ISSUE THREAD

**Issue by wytcsuch** [AlexeyAB/darknet#1180]
hello，everone
I'm training a model with 15 objects，my training dataset contain 9000 images with the size of 1024 × 1024.At the begining of training，the avg loss is more than 1300，after nearly 5000 iterations， it reduce to 2.5.But after that it's always dancing between 2.0 and 3.0，and now the iteration is 9000，and it seems that the avg loss is unlikely to reduce.should I stop training and make changes to the cfg file or keep going ?
In addition, -1.#IND00 is often present in Avg IOU and Class during training. Does this matter?

**Comment by AlexeyAB** [comment]
@wytcsuch Hi,
What mAP can you get? https://github.com/AlexeyAB/darknet#when-should-i-stop-training

**Comment by wytcsuch** [comment]
@AlexeyAB 
Hi,I got the mAP just now，it is  0.532781，it seems ok

 for thresh = 0.25, precision = 0.85, recall = 0.29, F1-score = 0.43
 for thresh = 0.25, TP = 16087, FP = 2785, FN = 39377, average IoU = 63.60 %
 mean average precision (mAP) = 0.532781, or 53.28 %

**Comment by wytcsuch** [comment]
@AlexeyAB 
In addition, I want to get the detection result of my test set and export it to the txt file. Each class is a txt file, which contains the name of the picture and the location of the target. What should I do?

**Comment by AlexeyAB** [comment]
@wytcsuch 

Create directory `/results/` near with executable `darknet` file. Put list of images to the `data/valid.txt` file, set in the `cfg/coco.data` file `valid=data/valid.txt`.


*  And run this command:
`./darknet detector valid cfg/coco.data cfg/yolov3cfg yolov3.weights`
And look at the `/results/` directory


* Or run this command:
`./darknet detector valid cfg/coco.data cfg/yolov3cfg yolov3.weights -dont_show -ext_output < data/valid.txt > result.txt`
And look at the `result.txt` file

**Comment by AlexeyAB** [comment]
> mean average precision (mAP) = 0.532781, or 53.28 %

It is a good mAP. So it doesn't matter that avg loss is high.

**Comment by wytcsuch** [comment]
@AlexeyAB 
Thank you very much for your answer
